.github/workflows/ctest.yaml

```diff
@@ -0,0 +1,35 @@
+# This is the testing workflow
+
+name: CTest
+
+# Controls when the action will run. Triggers the workflow on push or pull request events
+on: [push, pull_request]
```

Comment:
Maybe we could spare some github time by filtering the paths on which a pull request changes something. Like:
```yaml
pull_request:
    paths:
      - 'coupling/**'
      - 'simplemd/**'
```

Reply:
Okay I am adding paths-ignore and also a branches-ignore pattern. If you really want the CI to not run for some reason, then just flag your branch as work-in-progress by naming it `something-WIP` and it will be excluded automatically. (However I don't think its a problem to run the CI anyway on every push)